jbake-core/src/main/java/org/jbake/app/ContentStore.java

```diff
@@ -290,21 +283,23 @@ private void insertTemplatesSignature(String currentTemplatesSignature) {
         executeCommand(STATEMENT_INSERT_TEMPLATES_SIGNATURE, currentTemplatesSignature);
     }
 
-    private DocumentList<DocumentModel> query(String sql) {
+    private synchronized DocumentList<DocumentModel> query(String sql) {
         activateOnCurrentThread();
         OResultSet results = db.query(sql);
         return DocumentList.wrap(results);
     }
 
-    private DocumentList<DocumentModel> query(String sql, Object... args) {
+    private synchronized DocumentList<DocumentModel> query(String sql, Object... args) {
         activateOnCurrentThread();
         OResultSet results = db.command(sql, args);
         return DocumentList.wrap(results);
     }
 
-    private void executeCommand(String query, Object... args) {
+    private synchronized void executeCommand(String query, Object... args) {
         activateOnCurrentThread();
+        db.getTransaction().begin();
         db.command(query, args);
+        db.getTransaction().commit();
```

Comment:
Why use synchronize here? Instead, you could catch `OConcurrentModificationException` and retry.

See: https://orientdb.org/docs/3.0.x/general/Concurrency.html